test/test-helpers/mocked-context.ts

```diff
@@ -58,7 +58,7 @@ export function mockedContext(params?: MockedContextParams): Context<Env> {
             },
           },
           validateAuthenticationCode: {
-            mutate: async (code) => {
+            mutate: async ({ code, email, tenantId }: { code: string, email: string, tenantId: string }) => {
```

Comment:
I would always define the type above this, rather than straight after

TypeScript gets messy enough as it is :wink:   could be `ValidateAuthenticationCodeMutateParams`

Reply:
Yepp, would be better.

Reply:
Hmm.. This is mock (or rather a fixture). The actual types are created by Zod. I'll see if we can fetch this type dynamically instead.

Reply:
Wow.. typescript might not be intuitive but it's magic:
```
const caller = userRouter
  .createCaller({
    req: new Request("http://localhost:8787"),
    resHeaders: new Headers(),
    env: {},
    state: {
    } as DurableObjectState,
  })

type ValidateAuthenticationCodeParams = Parameters<typeof caller.validateAuthenticationCode>[0];
```